### ISSUE TITLE
"lesser" implementations

### ISSUE THREAD

**Issue by odilitime** [rosarior/Code-of-Merit#22]
Regarding: "Software is evolutive: the better implementations must supersede lesser implementations. Technical advantage is the primary evaluation metric."

In Engineering everything has pros and cons. Some fit the situation better than others. One algorithm is faster but takes more memory, another is slower but produces better output, this chunk uses lesser memory but makes the code harder to understand. One is lesser for certain situations and in another situation it isn't. Who's to determine what memory limits and what features should be traded for what speed? 

Because of that, I don't think there's an easy answer to this one beyond what ever the team thinks is best. Each project is going to have different values and places they fit into the eco-system, so implementations need to fit within that scope of values. Each proposal needs to be determined through rational discussion and benchmarks proving statements (along the lines of "Discuss or debate the idea, not the person.") because just because the project values are in one direction also don't always mean all patches need to be in that direction (sometimes balancing is needed as projects evolve).

I wish I had a more actionable request and less food for thought but hopefully we can all examine this together and brainstorm.

**Comment by nobledemon** [comment]
You're right, but that's why "evolutive" is the correct term. It implies adaptation, better means "more fit to the current scenario" and not just inherent superiority based on increasing the version number.

those are my two cents

**Comment by codergator** [comment]
> better implementations must supersede lesser implementations

I do agree that this might be too restricting on project owners. Because in engineering, there's hardly ever one "best". 

IMO the code should make it clear that _quality of the contribution_ is the important goal, while still putting all the power in those who own the project and have invested effort.

**Comment by codergator** [comment]
I'm also thinking that simply saying pursuit of merit, or pursuit of the best, says a lot. This is already staking a big claim in the ground. We believe that we won't always achieve it, but it's our goal. This contrasts with the CoC's we dislike which explicitly reject merit as a virtue.

**Comment by steveholden** [comment]
It's perhaps important to stress that merit should be a measure of the _contributions_ rather than of the _contributors_. The former should be judged rather than the latter - writing crappy code doesn't necessarily make you a crappy person, so while responses to contributions don't need to respect the code they should try to avoid disrespecting the person. "This code will not do" is a much better response than "you don't know what you are talking about" even when both responses might be true. I don't see any reason to make people feel bad about themselves just because they are on a learning curve, but neither do I feel that project maintainers must assume an obligation to teach (though many do).
